charts/supply-chain/templates/tasks/sign-artifact.yaml

```diff
@@ -0,0 +1,68 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: qtodo-sign-artifact
+  namespace: {{ .Values.global.namespace }}
+spec:
+  description: This Task can be used to sign the qtodo artifact.
+  workspaces:
+    - name: source
+      description: The workspace where the source code is cloned.
+  params:
+    - description: PEM encoded file containing the CA certificate to validate RHTAS services
+      name: ca-file
+      type: string
+    - description: Fulcio service URL
+      name: fulcio-url
+      type: string
+    - description: Rekor service URL
+      name: rekor-url
+      type: string
+    - description: TUF service URL
+      name: tuf-url
+      type: string
+    - description: Cosign CLI server URL
+      name: cli-server-url
+      type: string
+
+  volumes:
+    - name: spiffe-workload-api
+      csi:
+        driver: csi.spiffe.io
+        readOnly: true
+    - name: rhtas-functions
+      configMap:
+        name: rhtas-functions
+        defaultMode: 0755
+
+  steps:
+    - image: {{ .Values.tasks.images.cosign }}
+      name: sign-artifact
+      resources: {}
+      env:
+        - name: SPIFFE_ENDPOINT_SOCKET
+          value: /spiffe-workload-api/spire-agent.sock
+        - name: CLI_SERVER_URL
+          value: $(params.cli-server-url)
+        - name: TUF_URL
+          value: $(params.tuf-url)
+        - name: FULCIO_URL
+          value: $(params.fulcio-url)
+        - name: REKOR_URL
+          value: $(params.rekor-url)
+        - name: CA_FILE
+          value: $(params.ca-file)
+      volumeMounts:
+        - name: spiffe-workload-api
+          mountPath: /spiffe-workload-api
+          readOnly: true
+        - name: rhtas-functions
+          mountPath: /usr/local/lib
+          readOnly: true
+      script: |
+        #!/usr/bin/env bash
+
+        source /usr/local/lib/rhtas-functions
+
+        sign_artifact $(workspaces.source.path)/target/qtodo-*.jar
```

Comment:
Lets externalize the path to the artifact that we want to verify. Then in this line, we can perform a command similar to 

```
verify_artifact $(workspaces.source.path)/$(params.artifact)
```

We can still of course support wildcards as is currently configured